### PR TITLE
Change tests to reflect https://github.com/tc39/proposal-intl-list-format/pull/27 

### PR DESCRIPTION
**Pull request by FrankYFTang**
Based on 
https://github.com/tc39/proposal-intl-list-format/issues/16
https://github.com/tc39/proposal-intl-list-format/pull/27 change the spec so we need to change the test that 
1) either assert the throws of RangeError if 'style' is 'narrow' but 'type' is not set to 'unit', OR
2) change the test for 'style': 'narrow' for 'type':'unit'

### CODE REVIEW

**Comment by FrankYFTang** [comment]
@Ms2ger @leobalter

**Comment by littledan** [comment]
These tests look like a correct way to avoid the new error case. Are there tests that trigger the error as well?

**Comment by FrankYFTang** [comment]
hum.... I added those on my v8 tests but forget to put one to the test262 tests. I will add one but how should I do that with PR? I can push to the branch but should I create another PR for that?

**Comment by FrankYFTang** [comment]
ok, I push it and that now made into this PR. see the last three lines of 
test/intl402/ListFormat/constructor/constructor/options-style-valid.js

**Comment by FrankYFTang** [comment]
I don't see the green "MERGE" button in my account. Someone else have to push that for me. Thanks

**Comment by littledan** [comment]
@rwaldron @leobalter Would you mind if I merged this PR?

File: test/intl402/ListFormat/constructor/constructor/options-style-valid.js [hunk 1]
```diff
@@ -8,19 +8,31 @@ info: |
...
 }
+
+const lf = new Intl.ListFormat([], {"style": "narrow", "type": "unit"});
+const resolvedOptions = lf.resolvedOptions();
+assert.sameValue(resolvedOptions.style, "narrow");
+
+/*---
+    14. If style is "narrow" and type is not "unit", throw a RangeError exception.
+features: [Intl.ListFormat]
+---*/
```
Comment (Ms2ger) [inline test/intl402/ListFormat/constructor/constructor/options-style-valid.js]: Please remove this block; I expect it causes this lint error:
```
/home/travis/build/tc39/test262/test/intl402/ListFormat/constructor/constructor/options-style-valid.js: FRONTMATTER - No valid YAML-formatted frontmatter
```

File: test/intl402/ListFormat/constructor/constructor/options-style-valid.js [hunk 2]
```diff
@@ -29,10 +29,6 @@ const lf = new Intl.ListFormat([], {"style": "narrow", "type": "unit"});
 const resolvedOptions = lf.resolvedOptions();
 assert.sameValue(resolvedOptions.style, "narrow");
 
-/*---
-    14. If style is "narrow" and type is not "unit", throw a RangeError exception.
-features: [Intl.ListFormat]
```
Comment (FrankYFTang) [inline test/intl402/ListFormat/constructor/constructor/options-style-valid.js]: DONE

**Comment by FrankYFTang** [comment]
Pushed again. PTAL

**Comment by rwaldron** [comment]
@leobalter lint issue is resolved.

**Pull request merged**